geom/type_geometry.go

```diff
@@ -767,3 +767,26 @@ func (g Geometry) appendDump(gs []Geometry) []Geometry {
 	}
 	return gs
 }
+
+// DumpCoordinates returns the control points making up the geometry as a
+// Sequence.
+func (g Geometry) DumpCoordinates() Sequence {
+	switch g.gtype {
+	case TypeGeometryCollection:
+		return g.AsGeometryCollection().DumpCoordinates()
+	case TypePoint:
+		return g.AsPoint().DumpCoordinates()
+	case TypeLineString:
+		return g.AsLineString().Coordinates()
```

Comment:
There is no `DumpCoordinates` method for `LineString`, because the existing `Coordinates` method does the exact same thing. Even though it's a bit inconsistent, I think that's less confusing than having two methods that do exactly the same thing.

Reply:
if each geometry implemented `DumpCoordinates()`, could we do something like `g.AsDumpable().DumpCoordinates()` (for want of a better function name) and not required the switch-case block?

Reply:
That (or something similar) might be possible. The challenge would be implementing `AsDumpable()` on the `Geometry` type. The obvious way to implement it would be to use a `switch`, but that would defeat the point since we're trying to eliminate a switch statement. It _may_ be possible to use some sort of reflect/unsafe magic though.

The "switch" pattern actually shows up extremely frequently in this file (and is a bit of a code smell). Possibly it could be implemented generically so that we only need a single switch statement for all interface types.

Thanks for bringing this up! 😄 I've raised a ticket here: https://github.com/peterstace/simplefeatures/issues/399